### ISSUE TITLE
Missing mechanism to create apphook specific sitemap

### ISSUE THREAD

**Issue by moellering** [django-cms/django-cms#6680]
### Summary
When creating an app that is included on djangocms via apphooks at multiple places I don't see a way on how to provide a proper sitemap. 

### Expected behaviour
I should be able to provide a function for elements that provides a custom sitemap. 

I image something in the likes of the following
 
```
class MyApphook(CMSApp):
    name = "MyApphook"
    app_name = "myapphook"
    def get_urls(self, page=None, language=None, **kwargs):
        return ["myapphook.urls"]
    def get_sitemap(self, page=None, language=None, **kwargs):
        return MyApphookSitemap(page, language)
```

```
class MyApphookSitemap(CMSApp):
    def __init__(self, page, *args, **kwargs):
         super().__init__(*args,**kwargs)
         self.page = page
    changefreq = "monthly"
    priority = 0.5
    def items(self):
        return somehow_generate_the_entries()
    def location(self, obj):
        return somehow_generate_the_location_for(obj)
    def lastmod(self, obj):
        return somehow_calculate_the_lastmod_for(obj)
```

### Actual behaviour

The cms just creates an entry in the sitemap for the page that has the apphook. 
Problems:
a) The lastmod property does not reflect the content of the apphook, since it only checks for page properties
b) no sub pages of the apphook are included in the sitemap

### Environment

* Python version: 3.5.5
* Django version: 2.1.7
* django CMS version: 3.6.0

**Comment by SachaMPS** [comment]
You can do this in urls.py...

Example:
```
from cms.sitemaps import CMSSitemap
from apps.news.sitemaps import NewsSitemap

 sitemaps = {
        'cmspages': CMSSitemap,
        'news': NewsSitemap
}
urlpatterns += [
    url(r'^sitemap.xml$', sitemap, {'sitemaps': sitemaps}),
]
```

**Comment by moellering** [comment]
Hi Sacha,
thanks for your answer. Unfortunately that does not allow a developer of an application to modify the entry for the pages that are using the apphook. 
Once I create a page, set an apphook and publish the page, its url and modfication date are added to the apphook by the CMSSitemap. The content provided by the apphook might change every day, the i would not be able to modify the lastmod property.

Furthermore, the NewsSitemap would have to check if the pages would be reversable. This could be a problem for an app where the respective apphook is not yet assigned to a page.

Summing up: One would be able to generate a sitemap using the current mechanisms, but:
a) One has to use workarounds to update the lasmod property of pages with apphooks.
b) App Specific Sitemaps would have to search around which pages are using the respective apphooks or if the links are reversible.

I'm currently using my own Subclass of CMSSitemap to solve those problems which works fine for me, but maybe others might have a similar use case and may profit from a proper apphook centered interface here. Before creating a pull request, I'd like to hear your thoughts.
